core/commands/keystore.go

```diff
@@ -149,7 +149,28 @@ var KeyListCmd = &cmds.Command{
 		}
 
 		sort.Strings(keys)
-		res.SetOutput(&stringList{keys})
+
+		list := make([]string, 0, len(keys))
+
+		for _, key := range keys {
+			privKey, err := n.Repo.Keystore().Get(key)
+			if err != nil {
+				res.SetError(err, cmds.ErrNormal)
+				return
+			}
+
+			pubKey := privKey.GetPublic()
+
+			pid, err := peer.IDFromPublicKey(pubKey)
+			if err != nil {
+				res.SetError(err, cmds.ErrNormal)
+				return
+			}
+
+			list = append(list, key+" "+pid.Pretty())
```

Comment:
Using an array of strings here will make the json output of the api hard to parse. Lets make the output be an array of `KeyOutput` structs (which probably means defining a `KeyOutputList` struct because our use of the type system in the commands lib is weird).

Lets add a `-l` or `--show-ids` flag to the command and then in the output marshaler, then check that for whether or not to print the keys alongside the names.
Then in the text output marshaler, use a tabwriter to make sure the columns are aligned evenly when outputting both names and keys.